### DISCUSSION TITLE
[PSA] Sandbox with Typescript 4.5.0-beta and ts-node with nodenext modules

### DISCUSSION THREAD

**Discussion by FossPrime** [TypeStrong/ts-node#1492] [General]
Sandbox: https://codesandbox.io/s/ts-http-ygsxh?file=/tsconfig.json

```json-with-comments
// tsconfig.json
{
  "ts-node": {
    "files": true
  },
  "compilerOptions": {
    "target": "es2021",
    "module": "nodenext",
    "outDir": "/tmp/ts-http",
    "rootDir": "./src",
    "strict": true
  },
  "exclude": [
    "test"
  ]
}
```

```
// package.json
"type": "module",
"scripts": {
  "start": "nodemon -x node --loader ts-node/esm src/index.ts"
},
"dependencies": {
  "typescript": "4.5.0-beta",
  "@types/node": "14.14.37",
```

Though I'd post this somewhere... because it's not just a run ts-node-dev and you're done deal... they are on ts-node version 9, and don't override files/includes by default.

**Comment by cspotcode** [comment]
What's this for?

**Comment by FossPrime** [comment]
Nothing in particular... I guess anyone trying to upgrade to 4.5.0-beta to avoid module resolution issues and compiler time.

I added the definition and the untyped module just to make sure tsconfig was getting read properly... And as a playground of the many many ways to break this minimal app. We can close this... I was really hoping someone else had run into 4.5 issues and posted it here.
